control_panel_api/views.py

```diff
@@ -74,32 +110,65 @@ class AppS3BucketViewSet(viewsets.ModelViewSet):
 
     def perform_create(self, serializer):
```

Comment:
Should this wrapped in a transaction as well?

Reply:
Well spotted, thanks.